src/mono/wasm/Wasm.Build.Tests/Templates/InterpPgoTests.cs

```diff
@@ -0,0 +1,124 @@
+        {
+            // We need to invoke Greeting enough times to cause BCL code to tier so we can exercise interpreter PGO
+            // Invoking it too many times makes the test meaningfully slower.
+            const int iterationCount = 70;
+
+            string id = $"browser_{config}_{GetRandomId()}";
+            _testOutput.WriteLine("/// Creating project");
+            string projectFile = CreateWasmTemplateProject(id, "wasmbrowser");
+
+            _testOutput.WriteLine("/// Updating JS");
+            UpdateBrowserMainJs((js) => {
+                // We need to capture INTERNAL so we can explicitly save the PGO table
+                js = js.Replace(
+                    "const { setModuleImports, getAssemblyExports, getConfig } = await dotnet",
+                    "const { setModuleImports, getAssemblyExports, getConfig, INTERNAL } = await dotnet"
+                );
+                // Enable interpreter PGO + interpreter PGO logging + console output capturing
+                js = js.Replace(
+                    ".create()",
+                    ".withConsoleForwarding().withElementOnExit().withExitCodeLogging().withExitOnUnhandledError().withRuntimeOptions(['--interp-pgo-logging']).withInterpreterPgo(true).create()"
+                );
+                // Call Greeting in a loop to exercise enough code to cause something to tier,
+                //  then call INTERNAL.interp_pgo_save_data() to save the interp PGO table
+                js = js.Replace(
+                    "const text = exports.MyClass.Greeting();",
+                    "let text = '';\n" +
+                    $"for (let i = 0; i < {iterationCount}; i++) {{ text = exports.MyClass.Greeting(); }};\n" +
+                    "await INTERNAL.interp_pgo_save_data();"
+                );
+                return js;
+            }, DefaultTargetFramework);
+
+            _testOutput.WriteLine("/// Building");
+
+            new DotNetCommand(s_buildEnv, _testOutput)
+                    .WithWorkingDirectory(_projectDir!)
+                    .Execute($"build -c {config} -bl:{Path.Combine(s_buildEnv.LogRootPath, $"{id}.binlog")}")
+                    .EnsureSuccessful();
+
+            _testOutput.WriteLine("/// Starting server");
+
+            // Create a single browser instance and single context to host all our pages.
+            // If we don't do this, each page will have its own unique cache and the table won't be loaded.
+            using var runCommand = new RunCommand(s_buildEnv, _testOutput)
+                                        .WithWorkingDirectory(_projectDir!);
+            await using var runner = new BrowserRunner(_testOutput);
+            var url = await runner.StartServerAndGetUrl(runCommand, $"run --no-silent -c {config} --no-build --project {projectFile} --forward-console");
+            var browser = await runner.SpawnBrowserAsync(url);
+            var context = await browser.NewContextAsync();
```

Comment:
nit: Suggest using specific types because these aren't common to WBT, and seeing the types clearly would help with reading them.
```suggestion
            IBrowser browser = await runner.SpawnBrowserAsync(url);
            IBrowserContext context = await browser.NewContextAsync();
```